### PR TITLE
feat: add simulator test wiring

### LINKED ISSUE #11: feat: simulation support
Add simulator support for the TokenFactory module.

### PR DESCRIPTION
**Pull request by fmorency**
Fixes #11 

## Summaryt
This PR fixes the existing simulator cases and adds simulator test wiring.

All tests pass except `TestAppImportExport` like in https://github.com/strangelove-ventures/poa/pull/186

Not sure why.

# Run

```bash
go test -timeout 3000s -run ^TestFullAppSimulation$ github.com/strangelove-ventures/tokenfactory/app -v -Enabled -Commit=true
```

### CODE REVIEW

**Comment by fmorency** [comment]
@Reecepbcups I can't assign you as a reviewer. Commenting for visibility.

File: app/sim_test.go [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+func init() {
+	simcli.GetSimulatorFlags()
+	flag.BoolVar(&FlagEnableStreamingValue, "EnableStreaming", false, "Enable streaming service")
+}
+
+// fauxMerkleModeOpt returns a BaseApp option to use a dbStoreAdapter instead of
+// an IAVLStore for faster simulation speed.
+func fauxMerkleModeOpt(bapp *baseapp.BaseApp) {
+	bapp.SetFauxMerkleMode()
+}
```
Comment (Reecepbcups) [inline app/sim_test.go:51]: not used?

File: app/sim_test.go [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+
+// interBlockCacheOpt returns a BaseApp option function that sets the persistent
+// inter-block write-through cache.
+func interBlockCacheOpt() func(*baseapp.BaseApp) {
+	return baseapp.SetInterBlockCache(store.NewCommitKVStoreCacheManager())
+}
```
Comment (Reecepbcups) [inline app/sim_test.go:57]: not used?

File: app/sim_test.go [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+		simtypes.RandomAccounts, // Replace with own random account function if using keys other than secp256k1
+		simtestutil.SimulationOperations(app, app.AppCodec(), config),
+		BlockedAddresses(),
+		config,
+		app.AppCodec(),
+	)
+
+	// export state and simParams before the simulation error is checked
+	err = simtestutil.CheckExportSimulation(app, config, simParams)
+	require.NoError(t, err)
```
Comment (Reecepbcups) [inline app/sim_test.go:96]: I am getting a `invalid height: 2; expected: 1` error on `tokenfactory/app/sim_test.go:98` for all the test

```
Starting the simulation from time Sun May 21 15:21:12 UTC 2248 (unixtime 8785034472)
Simulating... block 1/500, operation 300/332.     sim_test.go:99: 
                Error Trace:    /home/reece/Desktop/Programming/Go/tokenfactory/app/sim_test.go:99
                Error:          Received unexpected error:
                                invalid height: 2; expected: 1
```

`go test -timeout 3000s -run ^TestFullAppSimulation$ github.com/strangelove-ventures/tokenfactory/app -v -Enabled`

Maybe something with `config.InitialBlockHeight`  ? or if simErr needs to just be checked selectively if some simParams or app value is set

**Review by Reecepbcups — CHANGES_REQUESTED** [review]

File: app/sim_test.go (hunk 3 above)
Comment (fmorency) [reply app/sim_test.go:96]: This is a known CosmosSDK issue. See https://github.com/cosmos/cosmos-sdk/issues/20592.

Try running with `-Commit=true`

Comment (Reecepbcups) [reply app/sim_test.go:96]: cool ty! updated the summary. Leaving this as unresolved for future context

**Review by Reecepbcups — APPROVED** [review]

**Comment by Reecepbcups** [comment]
Thanks!

**Pull request merged**